docs/migration-service.md

```diff
@@ -122,96 +201,258 @@ flowchart TD
 
-#### Cryptography
-A pair-wise key establishment scheme can be leveraged to establish a symmetric key. A suitable implementation can be found in [libsodium](https://doc.libsodium.org/) and its rust wrapper [libsodium-rs](https://docs.rs/crate/libsodium-rs/latest). Specifically, the **key exchange protocol based on X25519** would be a good fit for the given set-up (c.f. [libsodium](https://doc.libsodium.org/key_exchange) and [libsodium-rs](https://docs.rs/libsodium-rs/latest/libsodium_rs/crypto_kx/index.html)).
+```Rust
+/// Backup service authentication information (PLANNED extension for hard launch)
+pub struct BackupServiceInfo {
+    /// Ed25519 public key for mutual TLS authentication
+    pub public_key: Ed25519PublicKey,
+    
+    /// TEE information including attestation
+    /// Soft launch: mocked attestations accepted; Hard launch: real attestations required
+    /// Reuses existing TeeState struct and verification logic from the MPC contract
+    pub tee_state: TeeState,
+}
+```
 
-The key establishment scheme requires that the node as well as the backup service have *mutually authenticated Curve25519 public keys*. The NEAR blockchain can be leveraged for this, more specifically, for each backup generation or recovery, the node and the backup service generate ephemeral keys on the `Curve25519` and publish them on the MPC smart contract. The node and the backup service can then each run a key generation protocol using their private key and the public key of the other party. 
+The backup service attestation verification would follow the same process as MPC node attestations:
+1. Backup service generates TLS keypair inside TEE
+2. Backup service generates account keypair inside TEE for signing contract transactions (required to submit the attestation to the contract)
+3. Creates `ReportData` V1: `[version(2 bytes big endian) || sha384(TLS pub key || account_pubkey) || zero padding]`
+4. Obtains TEE quote embedding the `ReportData`
+5. Submits attestation via `register_backup_service(tls_public_key, account_public_key, attestation)`
+6. Contract verifies (using existing `TeeState` verification logic):
+   - Quote validity via attestation provider
+   - Docker image hash against allowed list
+   - Launcher compose hash (if applicable)
+   - Timestamp within deadline
+   - `ReportData` matches SHA3-384 hash of `SHA3-384(tls_public_key || account_public_key)`
+   - Transaction signer's public key matches `account_public_key` via `env::signer_account_pk()`
+7. Contract stores attestation in `backup_services_info[AccountId].tee_state` (reusing existing `TeeState` struct and verification logic)
+
+> **Note**: Unlike MPC nodes which may need multiple attestations per operator, backup services use a simpler one-per-operator model. The `AccountId` remains the unique identifier, consistent with soft launch.
+
+#### Migration Methods
```

Comment:
We need to add methods for voting on docker & launcher image hashes.
We probably have options:
- extend (i.e. through enum) the existing endpoints voting for MPC image hashes
- create new endpoints
Probably the latter is preferred, as we don't want any changes to the backup service breaking backwards compatibility with the MPC node.